lib/ansible/modules/cloud/openstack/os_port.py

```diff
@@ -228,21 +239,35 @@ def _needs_update(module, port, cloud):
                       'device_id',
                       'binding:vnic_type',
                       'port_security_enabled']
-    compare_dict = ['allowed_address_pairs',
-                    'extra_dhcp_opts']
+    compare_list_dict = ['allowed_address_pairs',
+                         'extra_dhcp_opts']
     compare_list = ['security_groups']
 
     for key in compare_simple:
         if module.params[key] is not None and module.params[key] != port[key]:
             return True
-    for key in compare_dict:
-        if module.params[key] is not None and module.params[key] != port[key]:
-            return True
     for key in compare_list:
         if module.params[key] is not None and (set(module.params[key]) !=
                                                set(port[key])):
             return True
 
+    for key in compare_list_dict:
+        if module.params[key] is not None:
```

Comment:
unless we need this to be specifically a check that the option *is not None* I think we should do something like this:

``` python
if not module.params[key]:
```

Reply:
I agree, I honestly cannot remember why I explicitly checked for None